.github/workflows/main.yml

```diff
@@ -1,79 +1,256 @@
-name: Main
+name: Continuous Integration
 
 on:
   push:
-    branches: master
+    branches:
+      - master
   pull_request:
-    branches: master
+    branches:
+      - master
 
 jobs:
-  Tests_pandas:
-    name: Tests pandas / files
-    runs-on: ${{ matrix.os }}
-    env:
-      BACKENDS: "pandas csv parquet hdf5"
-    strategy:
-      fail-fast: false
-      matrix:
-        os: [ubuntu-latest, windows-latest]
-        python_version: ["3.7", "3.9"]
+  check_setup_py:
+    runs-on: ubuntu-latest
     steps:
-    - name: checkout
-      uses: actions/checkout@v1
-
-    - name: set up environment
-      run: ./ci/setup_env.sh "${{ matrix.python_version }}" "$BACKENDS"
-      shell: bash
+      - name: checkout
+        uses: actions/checkout@v2
 
-    - name: run tests
-      run: PYTEST_BACKENDS=$BACKENDS ./ci/run_tests.sh --numprocesses auto
-      shell: bash
+      - name: install python
+        uses: actions/setup-python@v2
 
-    - name: pytest errors
-      run: ./ci/pytest_errors.sh
-      if: failure()
+      - name: install poetry
+        run: pip install poetry-core black tomli
 
-  Tests_dask:
-    name: Tests dask
+      - name: check whether setup.py needs to be regenerated
+        env:
+          # this environment variable is set is to ensure reproducible setup.py
+          # generation
+          #
+          # because various data structures in poetry are unordered, an can
+          # result in things like package extras appearing in a different order
+          PYTHONHASHSEED: 0
```

Comment:
Can we incorporate this into the script?

Reply:
Unfortunately no, because this variable appears to have no effect if set before startup.